### PR TITLE
Solana deposit and call

### PR DESCRIPTION
**Pull request by fadeev**
* Import IDL from npm
* Separate `deposit` and `depositAndCall` functions.

## Summary by CodeRabbit

- **New Features**
	- Added a new method for enhanced Solana transaction capabilities.
	- Introduced a new Hardhat task for executing deposit and call operations from the command line.
	- New dependency added for Solana protocol contracts.

- **Bug Fixes**
	- Streamlined the deposit function by removing unnecessary parameters for a simpler interface.

- **Chores**
	- Updated dependency versions in the package configuration.
	- Ensured proper formatting in configuration files.

### CODE REVIEW

**Comment by fadeev** [comment]
```
npx hardhat solana-deposit --amount 0.01 --recipient 0x2DCB13e7Eb5253fD5255Ce3CbCB199B48A0C7dD9
```

```
Transaction failed: TypeError: src.reduce is not a function
    at Sequence.encode (/Users/fadeev/github.com/zeta-chain/toolkit/node_modules/buffer-layout/lib/Layout.js:1112:22)
    at Structure.encode (/Users/fadeev/github.com/zeta-chain/toolkit/node_modules/buffer-layout/lib/Layout.js:1263:26)
    at WrappedLayout.encode (/Users/fadeev/github.com/zeta-chain/toolkit/node_modules/@coral-xyz/borsh/src/index.ts:115:24)
    at Structure.encode (/Users/fadeev/github.com/zeta-chain/toolkit/node_modules/buffer-layout/lib/Layout.js:1263:26)
    at BorshInstructionCoder.encode (/Users/fadeev/github.com/zeta-chain/toolkit/node_modules/@coral-xyz/anchor/src/coder/borsh/instruction.ts:63:32)
    at /Users/fadeev/github.com/zeta-chain/toolkit/node_modules/@coral-xyz/anchor/src/program/namespace/index.ts:62:43
    at MethodsBuilder.ix [as _ixFn] (/Users/fadeev/github.com/zeta-chain/toolkit/node_modules/@coral-xyz/anchor/src/program/namespace/instruction.ts:60:15)
    at MethodsBuilder.instruction (/Users/fadeev/github.com/zeta-chain/toolkit/node_modules/@coral-xyz/anchor/src/program/namespace/methods.ts:354:17)
    at async ZetaChainClient.solanaDeposit (/Users/fadeev/github.com/zeta-chain/toolkit/packages/client/src/solanaDeposit.ts:90:32)
    at async SimpleTaskDefinition.main [as action] (/Users/fadeev/github.com/zeta-chain/toolkit/packages/tasks/src/solanaDeposit.ts:30:3)
```

**Comment by fadeev** [comment]
Deposit and call:

```
npx hardhat solana-deposit-and-call --amount 0.01 --recipient 0x0b28dd447932003D40563B0e3707ab3b80a4d956 --types '["address", "address"]' 0x05BA149A7bd6dC1F937fA9046A9e05C05f3b18b0 0xe7508B5026f032b37663718192bA63a40954F2c0
```

https://zetachain-athens.blockpi.network/lcd/v1/public/zeta-chain/crosschain/inboundHashToCctxData/vPCwxf9bxP9vVHpdRHi6CBPnrDuKUxVn6zFGJctiRchvdwt9qhsnvD9eLUFXsz6NPVJhFe7XMXu5A9soqC4HVaC

It's failing, but that's expected, because currently Solana routes calls to `onCrossChainCall`, but the contract above uses the new `onCall`.

File: packages/client/src/solanaDepositAndCall.ts [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    params: any[];
+    recipient: string;
+  }
+) {
+  if (!this.isSolanaWalletConnected()) {
+    throw new Error("Solana wallet not connected");
+  }
+
+  const network = "solana_" + this.network;
+  const api = getEndpoints("solana" as any, network);
```
[the remark this answers is not in the document]
Comment (fadeev) [reply packages/client/src/solanaDepositAndCall.ts:25]: https://github.com/zeta-chain/networks/pull/61

File: packages/tasks/src/solanaDepositAndCall.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      recipient = utils.solidityPack(
+        ["bytes"],
+        [utils.toUtf8Bytes(args.recipient)]
+      );
+    }
+  } catch (e) {
+    recipient = args.recipient;
+  }
+  const { amount, idPath } = args;
+  const params = [JSON.parse(args.types), args.values];
```
[the remark this answers is not in the document]
Comment (fadeev) [reply packages/tasks/src/solanaDepositAndCall.ts]: 1ec3efc

**Review by andresaiello — APPROVED** [review]

**Pull request merged**